sound/soc/intel/boards/Kconfig

```diff
@@ -615,6 +615,8 @@ config SND_SOC_INTEL_SOF_DA7219_MAX98373_MACH
 
 endif ## SND_SOC_SOF_JASPERLAKE
 
+if SND_SOC_SOF_HDA_LINK
+
```

Comment:
Question: config `SND_SOC_INTEL_SOF_SSP_AMP_MACH` is for SOF with amplifiers in `I2S` Mode, why does it depend on config `SND_SOC_SOF_HDA_LINK`,  which is documented as SOF support for `HDA Links`(HDA/HDMI)`?

Is it because SND_SOC_INTEL_SOF_SSP_AMP_MACH selects SND_SOC_HDAC_HDMI and SND_SOC_INTEL_HDA_DSP_COMMON

Reply:
it's a good question @aiChaoSONG 

In reality it provides a shortcut to make this machine driver conditional on cAVS+ platforms with SOF. This is consistent with other cases in the same Kconfig file where we use this, sometimes along with SOF_BAYTRAIL when the machine driver can also support legacy platforms.

It really saves us the hassle of listing which platforms are supported.